File: stress/t/arr_matmul3.f
      PROGRAM ARRMATMUL3
      INTEGER*8 N,R,C,K,ACC,A,B,T,S
      INTEGER*8 AR(0:8),BR(0:8),CR(0:8)
      N=3
      R=0
   10 IF (R.GE.N) GOTO 30
      C=0
   20 IF (C.GE.N) GOTO 25
      AR(R*N+C)=R*N+C+1
      IF (R.EQ.C) THEN
        BR(R*N+C)=1
      ELSE
        BR(R*N+C)=0
      ENDIF
      C=C+1
      GOTO 20
   25 CONTINUE
      R=R+1
      GOTO 10
   30 CONTINUE
      R=0
   40 IF (R.GE.N) GOTO 70
      C=0
   50 IF (C.GE.N) GOTO 65
      ACC=0
      K=0
   55 IF (K.GE.N) GOTO 60
      A=AR(R*N+K)
      B=BR(K*N+C)
      ACC=ACC+A*B
      K=K+1
      GOTO 55
   60 CONTINUE
      CR(R*N+C)=ACC
      C=C+1
      GOTO 50
   65 CONTINUE
      R=R+1
      GOTO 40
   70 CONTINUE
      S=0
      R=0
   80 IF (R.GE.N) GOTO 100
      C=0
   90 IF (C.GE.N) GOTO 95
      T=CR(R*N+C)
      S=S+T
      C=C+1
      GOTO 90
   95 CONTINUE
      R=R+1
      GOTO 80
  100 CONTINUE
      WRITE(*,'(A,I0)') 'trace_sum=',S
      T=CR(4)
      WRITE(*,'(A,I0)') 'c11=',T
      END

File: stress/t/arr_matmul3.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t n,r,c,k,acc,a,b,t,s;
  int64_t A[9],B[9],C[9];
  n=3;
  for(r=0;r<n;r++){
    for(c=0;c<n;c++){
      A[r*n+c]=r*n+c+1;
      if(r==c){ B[r*n+c]=1; } else { B[r*n+c]=0; }
    }
  }
  for(r=0;r<n;r++){
    for(c=0;c<n;c++){
      acc=0;
      for(k=0;k<n;k++){ a=A[r*n+k]; b=B[k*n+c]; acc=acc+a*b; }
      C[r*n+c]=acc;
    }
  }
  s=0;
  for(r=0;r<n;r++){ for(c=0;c<n;c++){ t=C[r*n+c]; s=s+t; } }
  printf("trace_sum="); printf("%ld",(long)s); printf("\n");
  t=C[4]; printf("c11="); printf("%ld",(long)t); printf("\n");
  return 0;
}
